### PR TITLE
fix(discover): dedupe LB trending in-loop so the queue deck can't get duplicate keys

### PR DESCRIPTION
**Pull request by skylerwshaw**
Fixes the `/discover` freeze in #147.

## What

Three lines:

- `exclude.add(rg_mbid.lower())` inside the ListenBrainz trending loops in `get_trending_filler` (`queue_strategies.py`) and `_build_anonymous_queue` (`queue_service.py`).
- Index-suffix the `{#each}` key in `DiscoverQueueDeck.svelte:591`.

## Why

`DiscoverQueueDeck.svelte:591` keys its `{#each}` on `release_group_mbid`. A duplicate makes Svelte 5 throw `each_key_duplicate`, which aborts the `/discover` render, so the skeletons never resolve and the SPA half-freezes with nav updating the URL but not the view. That matches the reports in #147, including "the webpage actually freezes, clicking buttons will no longer go to that page".

`8b7e68b` index-suffixed the two `discover/+page.svelte` keys. Line 591 is the remaining keyed `{#each}` on that route with no uniqueness guarantee behind it.

Duplicates can reach that key. Both trending paths append from `get_sitewide_top_release_groups` while testing only against `exclude`, which is built before the loop and never added to inside it:

```python
exclude = ignored_mbids | library_mbids | (seen_mbids or set())   # never mutated below
for rg in rgs:
    if not rg_mbid or rg_mbid.lower() in exclude:
        continue
    wildcards.append(...)          # a repeat in one response gives two identical keys
```

Nothing downstream catches it. Wildcards skip `round_robin_dedup_select`, because `_build_personalized_queue` dedupes only the personalised tier before merging via `interleave_at_positions`, and `_build_anonymous_queue` never calls round-robin at all. Empty mbids are already guarded at all 11 append sites, so only repeats get through.

The `exclude.add()` call is exactly what the MusicBrainz decade fallback already does at `queue_strategies.py:466` in the same function. The LB branch just omits it.

Both affected paths are ListenBrainz calls, and #147 has reports of the crash coinciding with ListenBrainz load, but I want to be careful about how far I push that. I have not shown that ListenBrainz actually returns repeated release groups, so I am not claiming this is definitely what those reporters are hitting.

The defect stands on its own either way: the queue treats an external response as unique and fails fatally rather than cosmetically when it is not.

One judgement call worth flagging: the frontend index-suffix is belt-and-braces once the backend dedupes, and it means deck components re-key as the queue advances. It is consistent with what `8b7e68b` did to the sibling keys, but if you would rather avoid the re-keying, dropping that line and keeping the two backend ones still fixes the bug.

## Testing

- [x] I have tested these changes locally
- [x] I have added or updated tests

`test_dedup_repeated_release_group` asserts a repeated release group yields one item. Confirmed it fails without the `exclude.add()` change (`assert ['dupe-1', 'dupe-1'] == ['dupe-1']`).

`DiscoverQueueDeck.svelte.spec.ts` passes (8 tests). Backend suite: 3160 passed across `tests/services/` and `tests/routes/`. Three failures in `test_wanted_watcher_service.py` are pre-existing and unrelated, confirmed identical on a clean `main` checkout.

I did not run `ruff format`, since the two backend files already drift from it on `main` and `backend-lint` only runs `ruff check`. Reformatting would have buried a three-line fix in a large unrelated diff.

## AI-Assisted Contributions

Claude Code wrote all three source changes and the test, and did the analysis of the dedupe paths. I reviewed it and verified the behaviour myself.

Worth knowing for calibration: it took several wrong attempts at this before landing here. It first blamed the `discover/+page.svelte` keys, then blamed null MBIDs reaching the queue deck, both from reading code rather than running it. The null theory was wrong, since `release_group_mbid` is a required `str` and every append site guards falsiness. Only the duplicate path holds up, and it is the one covered by the failing-without-the-fix test above. Reviewer scrutiny on the analysis in the "Why" section is welcome, though the test result stands on its own.

## Checklist

- [x] Backend lint passes (`make backend-lint`)
- [x] Backend tests pass (`make backend-test`)
- [x] Frontend lint passes (`make frontend-lint`)
- [x] Frontend type check passes (`make frontend-check`, 4881 files, 0 errors, 0 warnings)
- [x] No `any` in TypeScript
- [x] No hardcoded colors (design tokens only)
- [x] All external API calls have timeouts
- [x] New msgspec structs follow existing patterns (none added)
- [x] TanStack Query used for all new data fetching (none added)
- [x] No secrets, tokens, or credentials in source

### CODE REVIEW

**Comment by skylerwshaw** [comment]
Thanks, both points were fair. Pushed changes for each.

**Index-suffix key.** You are right that it was purely defensive and that re-keying on queue advance is a real cost. I have dropped it, so `DiscoverQueueDeck.svelte` is now untouched by this PR.

I did want to keep a guard though, because the backend guarantee is an invariant threaded across `_build_personalized_queue` → `_get_wildcard_albums` → `get_trending_filler` plus the top-up path, not something enforced where the key is consumed. This bug is itself evidence that class of invariant can break silently: the `exclude.add()` was already missing from those two loops. And the failure mode is asymmetric, since a repeat throws and takes down the whole route rather than rendering one album twice.

So the guard moved to the store instead, where it costs nothing at render time:

```ts
queue = dedupeByMbid(data.items.map((item) => ({ ...item })));
```

The key stays `item.release_group_mbid`, so nothing re-keys as the queue advances, and a duplicate cannot reach the `{#each}` regardless of what the backend sends. Used the `filter`/`findIndex` shape from `DiscoveryBatchModal.svelte:23` rather than a `Set`, since `svelte/prefer-svelte-reactivity` rejects a plain `Set` in a `.svelte.ts` file.

Worth flagging: this also covers the cache-restore path at line 242 (`queue = cached.data.items`), which bypasses `fetchQueue` entirely and so was not protected by the backend fix at all. That path had a latent issue too, since `currentIndex` was clamped against `cached.data.items.length - 1`; with dedupe shrinking the array that could point past the end, so it now clamps against `queue.length - 1`.

**Missing test for the anonymous path.** Fair. `_build_anonymous_queue` has no coverage today and `DiscoverQueueService` is never constructed anywhere in the suite, so testing it means standing up the first service-level fixture with 11 dependencies. That felt like more than this fix should carry, and CONTRIBUTING asks to keep PRs focused, so I have not done it here. Happy to add it in a follow-up if you would like it, or in this PR if you would rather it not ship uncovered.

What I did add instead is two store tests covering both dedupe sites, `drops a duplicate release group from a fetched queue` and `drops a duplicate release group from a cached queue and clamps the index`. Both verified to fail without the change.

All green: `make frontend-lint`, `make frontend-check` (4881 files, 0 errors), `make backend-lint`, backend `test_queue_strategies.py` 33 passed, store spec 10 passed, deck component spec 8 passed.

**Comment by skylerwshaw** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by skylerwshaw** [comment]
recheck

File: frontend/src/lib/components/discover/DiscoverQueueDeck.svelte [hunk 1]
```diff
@@ -588,7 +588,7 @@
 							>
 								Up next
 							</span>
-							{#each deck.queue as item, i (item.release_group_mbid)}
+							{#each deck.queue as item, i (`${item.release_group_mbid}-${i}`)}
```
[the remark this answers is not in the document]
Comment (skylerwshaw) [reply frontend/src/lib/components/discover/DiscoverQueueDeck.svelte]: Addressed in https://github.com/DroppedNeedle/DroppedNeedle/pull/262#issuecomment-5221975487

**Comment by HabiRabbu** [comment]
Manually merged into main through a4d205f; closing this PR.
